### DISCUSSION TITLE
Verifying commands being submitted for PRs

### DISCUSSION THREAD

**Discussion by FazleArefin** [tldr-pages/tldr#12361] [General]
How do maintainers usually verify the commands being submitted in PRs before approving them?

I usually spin up a docker container suitable for the command (rust:latest for rust/cargo specific command, node:latest for any node/npm specific command, etc); otherwise a vagrant box. Is there a _better_ way?

(The reason I am asking is because I have found a number of commands which are wrong and smoke test was probably not done before being committed)

**Comment by kbdharun** [comment]
We generally don't verify the commands locally in most cases (especially in PRs with multiple additions and for PRs in other platforms than the ones we have due to hardware constraints [i.e. I don't have a Mac nor use any BSD or Sun OS platforms so can't check any changes manually, so I am fully reliant on the documentation]).

We generally refer to the documentation/man page before approving a pull request which has been outdated in multiple cases, unfortunately, especially with larger utilities. While a smoke test could be done for every command/page submitted (for platforms that we use) it would be difficult with the number of active maintainers we have (and the effort it would take to install/verify them). This is a very good approach if you have time. But for now, if we find docs to be outdated, I would suggest contacting upstream and reporting this issue to them.
